CONTRIBUTING.md

```diff
@@ -29,16 +29,14 @@ You'll also need [Insta](https://insta.rs/docs/) to update snapshot tests:
 cargo install cargo-insta
 ```
 
-And you'll need pre-commit to run some validation checks:
-
-```shell
-pipx install pre-commit  # or `pip install pre-commit` if you have a virtualenv
-```
+You'll need [uv](https://docs.astral.sh/uv/getting-started/installation/) (or `pipx` and `pip`) to
+run Python utility commands.
 
 You can optionally install pre-commit hooks to automatically run the validation checks
 when making a commit:
 
 ```shell
+uv tool install pre-commit
 pre-commit install
```

Comment:
Shouldn't this be `uvx pre-commit install`, as it won't necessarily be the first result for pre-commit on the PATH?

Reply:
If you `uvx pre-commit install` won't the pre-commit executable be missing during the git hook invocations?

Reply:
Just tested it in a clean docker container, and `uvx pre-commit install` works. It uses the python that it was originally executed with, so it doesn't need to be in the PATH (although it falls back to that if the original python is no longer available for some reason).

Reply:
I assume this relies on an environment that's in the uv cache though — that's generally bad practice.